includes/wikia/models/DesignSystemGlobalNavigationModel.class.php

```diff
@@ -0,0 +1,320 @@
+					'value' => $userName
+				],
+				'url' => AvatarService::getAvatarUrl( $userName, 50 ),
+			],
+			'links' => [
+				[
+					'type' => 'link-text',
+					'href' => $this->getPageUrl( $userName, NS_USER ),
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-user-view-profile'
+					]
+				],
+				[
+					'type' => 'link-text',
+					'href' => $isMessageWallEnabled
+						? $this->getPageUrl( $userName, NS_USER_WALL )
+						: $this->getPageUrl( $userName, NS_USER_TALK ),
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => $isMessageWallEnabled
+							? 'global-navigation-user-message-wall'
+							: 'global-navigation-user-my-talk'
+					]
+				],
+				[
+					'type' => 'link-text',
+					'href' => $this->getPageUrl( 'Preferences', NS_SPECIAL ),
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-user-my-preferences'
+					]
+				],
+				[
+					'type' => 'link-text',
+					'href' => $this->getPageUrl( 'Contents', NS_HELP ),
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-user-help'
+					]
+				],
+				[
+					'type' => 'link-authentication',
+					'href' => $this->getPageUrl( 'UserLogout', NS_SPECIAL ),
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-user-sign-out'
+					],
+					'param-name' => 'returnto'
+				]
+			],
+		];
+	}
+
+	private function getNotifications( $user ) {
+		$userName = $user->getName();
+
+		return [
+			'header' => [
+				'type' => 'line-image',
+				'image' => 'wsd-icons-bell',
+				'title' => [
+					'type' => 'translatable-text',
+					'key' => 'global-navigation-notifications-title'
+				]
+			],
+			'module' => [
+				'type' => 'notifications',
+				'url' => $this->isMessageWallEnabled()
+					? $this->getPageUrl( $userName, NS_USER_WALL )
+					: $this->getPageUrl( $userName, NS_USER_TALK )
+			]
+		];
+	}
+
+	private function getVerticalsSection() {
+		return [
+			'links' => [
+				[
+					'type' => 'link-branded',
+					'brand' => 'tv',
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-fandom-overview-link-vertical-tv'
+					],
+					'href' => $this->getHref( 'tv' ),
+				],
+				[
+					'type' => 'link-branded',
+					'brand' => 'games',
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-fandom-overview-link-vertical-games'
+					],
+					'href' => $this->getHref( 'games' ),
+				],
+				[
+					'type' => 'link-branded',
+					'brand' => 'movies',
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-fandom-overview-link-vertical-movies'
+					],
+					'href' => $this->getHref( 'movies' ),
+				]
+			]
+		];
+	}
+
+	private function isMessageWallEnabled() {
+		return WikiFactory::getVarValueByName( 'wgEnableWallExt', $this->wikiId );
+	}
+
+	private function isCorporatePage() {
```

Comment:
What was the reason not to extend WikiaPageType::isCorporatePage() to accept an optional wiki_id?